homeassistant/components/backup/config.py

```diff
@@ -272,37 +279,48 @@ def _schedule_next(
         """Schedule the next backup."""
         self._unschedule_next(manager)
         now = dt_util.now()
-        seed_time = manager.config.data.last_automatic_backup or now
-        cron_event = CronSim(cron_pattern, seed_time)
+        if (cron_event := self.cron_event) is None:
+            seed_time = manager.config.data.last_automatic_backup or now
+            cron_event = self.cron_event = CronSim(cron_pattern, seed_time)
         next_time = next(cron_event)
 
         if next_time < now:
-            # schedule a backup at next daily time if we missed
-            # the last scheduled backup
+            # schedule a backup at next daily time once
+            # if we missed the last scheduled backup
             cron_event = CronSim(CRON_PATTERN_DAILY, now)
             next_time = next(cron_event)
+            # reseed the cron event attribute
+            # add a day to the next time to avoid scheduling at the same time again
+            self.cron_event = CronSim(cron_pattern, now + timedelta(days=1))
 
         async def _create_backup(now: datetime) -> None:
             """Create backup."""
             manager.remove_next_backup_event = None
             config_data = manager.config.data
-            config_data.last_automatic_backup = dt_util.now()
-            manager.config.save()
             self._schedule_next(cron_pattern, manager)
 
-            # Create the backup
-            await manager.async_create_backup(
-                agent_ids=config_data.create_backup.agent_ids,
-                include_addons=config_data.create_backup.include_addons,
-                include_all_addons=config_data.create_backup.include_all_addons,
-                include_database=config_data.create_backup.include_database,
-                include_folders=config_data.create_backup.include_folders,
-                include_homeassistant=True,  # always include HA
-                name=config_data.create_backup.name,
-                password=config_data.create_backup.password,
-            )
+            # create the backup
+            try:
+                await manager.async_create_backup(
+                    agent_ids=config_data.create_backup.agent_ids,
+                    include_addons=config_data.create_backup.include_addons,
+                    include_all_addons=config_data.create_backup.include_all_addons,
+                    include_database=config_data.create_backup.include_database,
+                    include_folders=config_data.create_backup.include_folders,
+                    include_homeassistant=True,  # always include HA
+                    name=config_data.create_backup.name,
+                    password=config_data.create_backup.password,
+                )
+            except Exception:  # noqa: BLE001
+                # another more specific exception will be added
+                # and handled in the future
+                LOGGER.exception("Unexpected error creating automatic backup")
```

Comment:
What does this comment mean exactly?

Reply:
This is my ongoing work. We'll add an exception that the BackupManager will raise for all exceptions coming from it. We'll catch that exception here besides broad exception. Known exceptions will just log an error without stack trace.